Refactor: Unify string concatenation style using fmt.Sprintf (#131566)

Sapthagiri777 opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Consider creating this PR as draft: https://github.com/kubernetes/kubernetes/blob/master/CONTRIBUTING.md#consider-opening-your-pull-request-as-draft
2. Follow the instructions for writing a release note from k8s: https://git.k8s.io/community/contributors/guide/release-notes.md
-->

### What this PR does
This PR refactors places where string concatenation was used to construct values,
replacing them with `fmt.Sprintf` for consistency and readability.

### Before this PR:
Multiple areas in test files used `+` for string construction.

### After this PR:
Replaced with `fmt.Sprintf`, improving clarity and style.


Fixes #131566

### Why we need it and why it was done in this way
Following the code style discussed in the issue, `fmt.Sprintf` is more consistent and readable,
especially when composing complex strings like JSON or logs.

### Special notes for your reviewer
This is a beginner-friendly contribution made by following Kubernetes code conventions.
Tested locally and verified the style change doesn’t affect logic.

### Checklist

- [x] PR description is clear
- [x] Changes are scoped and consistent
- [x] Manual tests done (no logic change, only style)
- [x] No unit tests needed for this string formatting change
- [x] DCO signoff is present

### Release note
```release-note
NONE

Sapthagiri777 committed:
Refactor: Unify string concatenation style using fmt.Sprintf (#131566)

Sapthagiri777 commented:
Hi @smarterclayton and @sergeykanzhelev 👋

Could you please take a moment to review and approve this PR?

It addresses issue #131566 by refactoring string concatenation to `fmt.Sprintf` style in several e2e files.

Thanks a lot for your time and help!

haircommander commented on test/e2e/apimachinery/resource_quota.go:
```diff
@@ -1035,7 +1036,7 @@ var _ = SIGDescribe("ResourceQuota", func() {
 		gomega.Expect(rq.Items).To(gomega.HaveLen(1), "Failed to find ResourceQuotes %v", rqName)
 
 		ginkgo.By("Patching the ResourceQuota")
-		payload := "{\"metadata\":{\"labels\":{\"" + rqName + "\":\"patched\"}},\"spec\":{\"hard\":{ \"memory\":\"750Mi\"}}}"
```
this likely doesn't have a huge effect, but fmt.Sprintf is less efficient than just concating the strings, as Sprintf has to parse the format. 

I do think not needing to escape " makes it more readable, but that could be done without Sprintf...

Sapthagiri777 replied:
Thanks @haircommander for the feedback!  
I understand that `fmt.Sprintf()` may not be optimal here. I'll refactor the string to use concatenation while keeping it readable.

Also, I'd really appreciate any suggestions on what else I could improve in this PR — especially regarding style, efficiency, or readability.  
I'm eager to learn and make meaningful contributions. Let me know if there's anything you'd recommend for next time or beyond!

Pushing an update shortly. 😊

Sapthagiri777 replied:
Thanks @haircommander for the feedback! I've updated that line to use raw string + concatenation as suggested.

Also, if you notice anything else across the PR that I could improve (in readability or consistency), I’d be happy to make adjustments.

Sapthagiri777 committed:
Revert to string concatenation for readability

chansuke commented on test/e2e/apimachinery/resource_quota.go:
```diff
@@ -17,6 +17,7 @@ limitations under the License.
 package apimachinery
 
 import (
+        "fmt"
```
Duplicated import here?

Sapthagiri777 replied:
I changed it .

chansuke commented on test/e2e/apps/controller_revision.go:
```diff
@@ -17,6 +17,7 @@ limitations under the License.
 package apps
 
 import (
+        "fmt"
```
same

Sapthagiri777 replied:
I changed it .

chansuke commented on test/e2e/common/storage/host_path.go:
```diff
@@ -17,6 +17,7 @@ limitations under the License.
 package storage
 
 import (
+        "fmt"
```
same

Sapthagiri777 replied:
I changed it .

chansuke reviewed (commented):
nit: Looks like `fmt` is imported twice in several files

Sapthagiri777 committed:
Remove duplicate fmt imports

Sapthagiri777 commented:
Thanks @haircommander for the feedback!

I understand that `fmt.Sprintf()` may not be optimal here. I've updated the PR to revert to string concatenation, keeping the string readable and avoiding unnecessary escapes.

Let me know if there's anything else I can improve in this PR — happy to iterate further!

Sapthagiri777 commented:
Thanks for pointing it out! I've removed the duplicate fmt imports.

Sapthagiri777 commented:
Hi @chansuke, I’ve addressed all the comments by removing the duplicate fmt imports. Please let me know if there’s anything else to improve. Thanks!

aaron-prindle commented:
@Sapthagiri777 currently the PR description mentions fixes https://github.com/kubernetes/kubernetes/issues/131566 but my understanding is that this PR is unrelated.  Can you confirm and link the associated issue this is related to?  Hoping to understand more about the concat -> fmt.Sprintf changes here and rationale.  Thanks

aaron-prindle commented:
/triage accepted

yongruilin commented:
This PR is unrelated to https://github.com/kubernetes/kubernetes/issues/131566. Could you update to link to the correct issue? thanks!
